src/link.cpp

```diff
@@ -898,10 +898,44 @@ namespace mamba
             return std::make_tuple(validate::sha256sum(dst), rel_dst);
         }
 
-        if (path_data.path_type == PathType::HARDLINK)
+        if (path_data.no_link || (path_data.path_type == PathType::HARDLINK))
```

Comment:
maybe change the order of comparison here because path_data.path_type == HARDLINK is much more likely (speed boost!)

Reply:
done ! huge boost ! :)